### PR TITLE
[New] `CalendarDay`: Add `aria-current` for "today"

### LINKED ISSUE #2009: CalendarDay is missing aria-current attribute for today's date
An accessibility attribute `aria-current="date"` is missing on the `CalendarDay` of today's date. It marks the current date to screen reader users.

https://www.w3.org/TR/wai-aria-1.1/#aria-current

### PR DESCRIPTION
**Pull request by mmarkelov**
close #2009

### CODE REVIEW

File: src/components/CalendarDay.jsx [hunk 1]
```diff
@@ -154,6 +154,7 @@ class CalendarDay extends React.PureComponent {
         role="button" // eslint-disable-line jsx-a11y/no-noninteractive-element-to-interactive-role
         ref={this.setButtonRef}
         aria-disabled={modifiers.has('blocked')}
+        {...(modifiers.has('today') ? { 'area-current': 'date' } : {})}
```
Comment (ljharb) [inline src/components/CalendarDay.jsx:157]:
```suggestion
        {...(modifiers.has('today') ? { 'aria-current': 'date' } : {})}
```

File: test/components/CalendarDay_spec.jsx [hunk 2]
```diff
@@ -54,6 +54,32 @@ describe('CalendarDay', () => {
       expect(wrapper.props().tabIndex).to.equal(tabIndex);
     });
 
+    describe('area-current', () => {
```
Comment (ljharb) [inline test/components/CalendarDay_spec.jsx:57]:
```suggestion
    describe('aria-current', () => {
```

File: test/components/CalendarDay_spec.jsx [hunk 3]
```diff
@@ -54,6 +54,32 @@ describe('CalendarDay', () => {
...
+      it('should add aria-current to date for today date', () => {
+        const modifiers = new Set(['today']);
+
+        const wrapper = shallow((
+          <CalendarDay
+            modifiers={modifiers}
+          />
+        )).dive();
+
+        expect(wrapper.prop('area-current')).to.equal('date');
```
Comment (ljharb) [inline test/components/CalendarDay_spec.jsx:67]:
```suggestion
        expect(wrapper.prop('aria-current')).to.equal('date');
```

File: test/components/CalendarDay_spec.jsx [hunk 4]
```diff
@@ -54,6 +54,32 @@ describe('CalendarDay', () => {
...
+      it('should not add aria-current for not today date', () => {
+        const modifiers = new Set();
+
+        const wrapper = shallow((
+          <CalendarDay
+            modifiers={modifiers}
+          />
+        )).dive();
+
+        expect(wrapper).to.not.have.property('area-current');
```
Comment (ljharb) [inline test/components/CalendarDay_spec.jsx:79]:
```suggestion
        expect(wrapper).to.not.have.property('aria-current');
```

File: test/components/CalendarDay_spec.jsx (hunk 3 above)
Comment (mmarkelov) [reply test/components/CalendarDay_spec.jsx]: @ljharb oops! my bad)

**Review by ljharb — APPROVED** [review]

**Pull request merged**